src/types/explorers.ts

```diff
@@ -0,0 +1,25 @@
+import { Chain } from "./general";
+
+export const explorerIdentifiers = ['etherscan', 'polygonscan', 'solscan'] as const;
+export type ExplorerIdentifier = typeof explorerIdentifiers[number];
+
+export interface ExplorerInfo {
+    /**
+     * Display name for explorer, ie: "Etherscan"
+     */
+    name: string;
+    addressUrlFormatter: (address: string) => string;
+}
+
+// ======= Chain-specific explorer types ======= //
+type ChainExplorer<C extends Chain, E extends ExplorerIdentifier> = Record<C, Record<Extract<ExplorerIdentifier, E>, ExplorerInfo>>;
+
+export type EthereumExplorer = ChainExplorer<'ethereum', 'etherscan'>;
+export type PolygonExplorer = ChainExplorer<'polygon', 'polygonscan'>;
+export type SolanaExplorer = ChainExplorer<'solana', 'solscan'>;
+
+/**
+ * Map of chain to all its explorers' info.
```

Comment:
this would be a map from chain to single explorer, right?

Reply:
negatory. MagicEden show 3 different explorer options, I see value in us treating explorers the same way as marketplaces, such that this PR would allow us to give a user options on which explorer they wanna use 
<img width="611" alt="image" src="https://user-images.githubusercontent.com/24663638/188290695-be5ae2a0-f614-42e2-ba59-c93fe235e470.png">